src/java_bytecode/java_string_library_preprocess.cpp

```diff
@@ -930,53 +930,72 @@ java_string_library_preprocesst::string_literal_to_string_expr(
 /// \param symbol_table: symbol table
 /// \return Code corresponding to:
 /// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-/// string_expr1 = {this->length; *this->data}
-/// string_expr2 = {arg->length; *arg->data}
-/// return cprover_string_equal(string_expr1, string_expr2)
+/// IF arg->@class_identifier == "java.lang.String"
+/// THEN
+///   string_expr1 = {this->length; *this->data}
+///   string_expr2 = {arg->length; *arg->data}
+///   return cprover_string_equal(string_expr1, string_expr2)
```

Comment:
I realise semtantically equiv but probably clearer if this comment includes the fact your introduce a temporary variable:
```
bool string_equals_tmp = cprover_string_equal(string_expr1, string_expr2);
return string_equals_tmp;
```